src/_posts/2020-01-14-fear-driven-development.md

```diff
@@ -0,0 +1,318 @@
+
+method howManyProductsAreOverOneHundredPounds() {
+    select all products where price is greater than £100
+    return how many products found 
+}
+```
+
+Now this choice means that the SQL required to execute this query is now spread over two places in
+the codebase. This will create many problems:
+
+* if the query is to be changed (e.g. schema changes), it has to be done in two places, and a
+developer won't necessarily know this is the case
+* who owns the data? Should the second method know how to get this information, or should it simply
+call another part of the system (e.g. `getProductsOverOneHundredPounds`) to get this
+information (not knowing or caring how this is done)? This can be achieved by modularising code in
+your codebase.
+
+### Tunnel-vision programming
+This infliction on a codebase can be found in both the aforementioned examples. It is simply when
+one is so focused on a particular task, that they do not step back to see the bigger picture. You
+are so concerned with implementing feature x, that you do not consider how it fits together with y
+and z.
+
+Tunnel-vision programming could be described as selfish, I might be tempted to define it as the
+below:
+
+> Satisfying one's immediate needs whilst degrading the health of the codebase
+
+Now, this can happen with or without fear. There are many different types of people working on
+codebases all around the world. Not all of them will be motivated to keep a codebase tidy, the
+decisions made might be driven by many factors, including fear.
+
+## 3. Where the fear stops
+Everything before now has provided context of how a codebase can be negatively impacted by fear,
+now I want to look at some techniques and tips for how we can avoid FDD being practised on our
+codebases.
+
+### Tests
+The title above is simple. Having meaningful tests can instill confidence that you are not breaking
+anything. If you are a practitioner of TDD, then this will also help you to have confidence, and
+most importantly for me, structure to how you work.
+
+If you want to come along and change the below:
+
+```
+method getProductsOverOneHundredPounds() {
+    select all products where price is greater than £100
+    return products found
+}
+```
+
+And in the codebase there is also a test for this:
+
+```
+method testForGetProductsOverOneHundredPounds() {
+    get result from getProductsOverOneHundredPounds
+    
+    result should equal expected result
+}
+```
+
+Then when you are going to make a change in `getProductsOverOneHundredPounds`, you will get quick
+feedback that something has been broken by the change you have made, because you will have run the
+tests (say it once more!). For instance, you might "accidentally" have changed the code to:
+
+```
+method getProductsOverOneHundredPounds() {
+    select all products where price is greater than £1000
+    return products found
+}
+```
+
+Fortunately your test might catch that extra `0` you added and you can revert your change. No fear.
```

Comment:
I would comment as well that tests are a great way to clarify context across team mates "is that what we want?" this benefit helped me a lot in my last client.

Reply:
This is a good point, I will take a look 👍 

Thank you for taking the time to suggest this.